src/Datadog.Trace/Logging/DatadogLogging.cs

```diff
@@ -0,0 +1,117 @@
+using System.Diagnostics;
+using System.IO;
+using Datadog.Trace.Vendoring.Serilog;
+using Datadog.Trace.Vendoring.Serilog.Events;
+using Datadog.Trace.Vendoring.Serilog.Sinks.File;
+
+namespace Datadog.Trace.Logging
+{
+    internal static class DatadogLogging
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/";
+        private static readonly string WindowsDefaultDirectory =
+            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+
+        private static readonly long? MaxLogFileSize = 10 * 1024 * 1024;
+        private static readonly LogEventLevel MinimumLogEventLevel = LogEventLevel.Verbose; // Lowest level
+
+        private static readonly ILogger SharedLogger = null;
+
+        static DatadogLogging()
+        {
+            // No-op for if we fail to construct the file logger
+            SharedLogger =
+                new LoggerConfiguration()
+                   .WriteTo.Sink<NullSink>()
+                   .CreateLogger();
+            try
+            {
+                var currentAppDomain = AppDomain.CurrentDomain;
+                var currentProcess = Process.GetCurrentProcess();
+
+                var debugEnabledVariable = Environment.GetEnvironmentVariable("DD_TRACE_DEBUG")?.ToLower();
+                if (debugEnabledVariable != "1" && debugEnabledVariable != "true")
+                {
+                    // No verbose or debug logs
+                    MinimumLogEventLevel = LogEventLevel.Information;
+                }
+
+                var maxLogSizeVar = Environment.GetEnvironmentVariable("DD_MAX_LOGFILE_SIZE");
+                if (!string.IsNullOrEmpty(maxLogSizeVar) && long.TryParse(maxLogSizeVar, out var maxLogSize))
+                {
+                    // No verbose or debug logs
+                    MaxLogFileSize = maxLogSize;
+                }
+
+                var nativeLogFile = Environment.GetEnvironmentVariable("DD_TRACE_LOG_PATH");
+                string logDirectory = null;
+
+                if (!string.IsNullOrEmpty(nativeLogFile))
+                {
+                    logDirectory = Path.GetDirectoryName(nativeLogFile);
+                }
+
+                if (logDirectory == null)
+                {
+                    if (Directory.Exists(WindowsDefaultDirectory))
+                    {
+                        logDirectory = WindowsDefaultDirectory;
+                    }
+                    else if (Directory.Exists(NixDefaultDirectory))
+                    {
+                        logDirectory = NixDefaultDirectory;
+                    }
+                    else
+                    {
+                        logDirectory = Environment.CurrentDirectory;
+                    }
+                }
+
+                // Ends in a dash because of the date postfix
+                var managedLogPath = Path.Combine(logDirectory, $"dotnet-tracer-{currentProcess.ProcessName}-.log");
```

Comment:
Looks like an extra `-` at the end.
```suggestion
                var managedLogPath = Path.Combine(logDirectory, $"dotnet-tracer-{currentProcess.ProcessName}.log");
```

Reply:
it's not an extra dash, the comment above explains it.

Reply:
Oops, reading comprehension fail. 🤦‍♂